redirect/routes.php

```diff
@@ -1,112 +1,86 @@
 <?php
 class Route {
 
-  private $ROUTES_JSON_STRING = '
-  {
-    "Special Stuff": {
-      "hunger": {
-        "description": "FMI Bistro Speiseplan",
-        "target": "http://tum.sexy/hunger",
-        "display": true
-      },
-      "rooms": {
-        "description": "MI Raumbelegungen",
-        "target": "http://tum.sexy/rooms",
-        "display": true
-      },
-      "c": {
-        "description": "TUM Online",
-        "target": "https://campus.tum.de/",
-        "display": true
-      },
-      "m": {
-        "description": "Moodle",
-        "target": "https://www.moodle.tum.de/",
-        "display": true
-      },
-      "stuff": {
-        "description": "Unistuff (ehemals Tumstuff)",
-        "target": "http://unistuff.org",
-        "display": true
-      },
-      "reddit": {
-        "description": "ReddiTUM",
-        "target": "https://reddit.com/r/redditum",
-        "display": true
-      },
-      "tedx": {
-        "description": "TEDxTUM Event-Seite",
-        "target": "http://tedxtum.com",
-        "display": false
-      }
-    },
-
-    "1. Semester": {
-
-    },
-
-    "2. Semester": {
-
-    },
-
-    "3. Semester": {
-
-    },
-
-    "4. Semester": {
-
-    },
-
-    "5. Semester": {
-      "numprog": {
-        "description": "Numerisches Programmieren",
-        "target": "http://www5.in.tum.de/wiki/index.php/Numerisches_Programmieren_-_Winter_15",
-        "display": true
-      }
-
-    },
-
-    "6. Semester": {
+  private $routes = [
+    'hunger'=> [
+      "description" => "FMI Bistro Speiseplan",
+      "target"=> "http://tum.sexy/hunger",
+      "display"=> true
+    ],
+    "rooms"=> [
+      "description"=> "MI Raumbelegungen",
+      "target"=> "http://tum.sexy/rooms",
+      "display"=> true
+    ],
+    "c"=> [
+      "description"=> "TUM Online",
+      "target"=> "https://campus.tum.de/",
+      "display"=> true
+    ],
+    "m"=> [
+      "description"=> "Moodle",
+      "target"=> "https://www.moodle.tum.de/",
+      "display"=> true
+    ],
+    "stuff"=> [
+      "description"=> "Unistuff (ehemals Tumstuff)",
+      "target"=> "http://unistuff.org",
+      "display"=> true
+    ],
+    "reddit"=> [
+      "description"=> "ReddiTUM",
+      "target"=> "https://reddit.com/r/redditum",
+      "display"=> true
+    ],
+    "tedx"=> [
+      "description"=> "TEDxTUM Event-Seite",
+      "target"=> "http://tedxtum.com",
+      "display"=> false
+    ],
+    "numprog"=> [
+      "description"=> "Numerisches Programmieren",
+      "target"=> "http://www5.in.tum.de/wiki/index.php/Numerisches_Programmieren_-_Winter_15",
+      "display"=> true
+    ]
+  ];
+
+  private $sections = [
+    "Special Stuff" => [
+      'hunger', 'rooms', 'c', 'm', 'stuff','reddit', 'tedx'
```

Comment:
This is the way you would build up the information in a database table. You would normally use a PK(int) for reference in order to be able to change the route, but this duplication is not so bad. Better than to have a single array which 'simulates' structure of data which have nothing in common.